ChattoAdditions/Source/Chat Items/CompoundMessage/Content/MessageContentFactoryProtocol.swift

```diff
@@ -21,49 +21,94 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-public final class MessageContentModule {
-    public typealias Presenter = Any
-    public let view: UIView
-    public let showBorder: Bool
-    public let presenter: Presenter
+import Chatto
+
+public protocol MessageContentPresenterProtocol {
+
+    var showBorder: Bool { get }
+
+    func contentWillBeShown()
+    func contentWasHidden()
+    func contentWasTapped()
+}
 
-    public init(view: UIView,
-                presenter: Presenter,
-                showBorder: Bool = false) {
-        self.view = view
+public final class DefaultMessageContentPresenter: MessageContentPresenterProtocol {
+
+    public init(showBorder: Bool) {
         self.showBorder = showBorder
-        self.presenter = presenter
     }
+
+    public let showBorder: Bool
+
+    public func contentWillBeShown() {}
+    public func contentWasHidden() {}
+    public func contentWasTapped() {}
 }
 
 public protocol MessageContentFactoryProtocol {
     associatedtype Model
-    func canCreateMessageModule(forModel model: Model) -> Bool
-    func createMessageModule(forModel model: Model) -> MessageContentModule
+    var identifier: String { get }
+    func canCreateMessageContent(forModel model: Model) -> Bool
+    func createNewMessageView() -> UIView
+    func createContentPresenter(forModel model: Model) -> MessageContentPresenterProtocol
+    func unbindContentPresenter(_ presenter: MessageContentPresenterProtocol)
+    func bindContentPresenter(_ presenter: MessageContentPresenterProtocol, withView view: UIView, forModel model: Model)
     func createLayoutProvider(forModel model: Model) -> MessageManualLayoutProviderProtocol
+    func createMenuPresenter(forModel model: Model) -> ChatItemMenuPresenterProtocol?
+}
+
+public extension MessageContentFactoryProtocol {
+    var identifier: String { return "\(type(of: self as Any))" }
```

Comment:
Could you try `String(describing: self)` instead?

Reply:
Replaced with `String(describing: type(of: self))` but I will have to doublecheck it works correctly without casting to `Any`.